tests/test_admin.py

```diff
@@ -173,7 +173,7 @@ def test_admin_form_edit_when_locked(self):
         self.snippet_version.publish(user=self.superuser)
         with self.login_user_context(self.superuser):
             edit_url = reverse("admin:djangocms_snippet_snippet_change", args=(self.snippet.id,),)
-            response = self.client.get(edit_url)
+            response = self.client.get(edit_url + "?q=read_only")
```

Comment:
We would need a separate test for this, by changing this test we lose the check on version locking and repurpose it for a new requirement for the preview read only functionality.